### DISCUSSION TITLE
(Help) modeling my database.

### DISCUSSION THREAD

**Discussion by brunowilliang** [pocketbase/pocketbase#1302] [Q&A]
I have a table of companies.

I need to add days of the week with specific times for each day. ex:

monday - closed
Tuesday - 09:00 to 18:00
Wednesday - 09:00 to 18:00
Thursday - 09:00 to 18:00
Friday - 09:00 to 18:00
Saturday - closed
sunday - closed

how would you model this in pocketbase?

**Comment by ganigeorgiev** [comment]
It depends how you are planning to use the time ranges, whether they are only for display purposes, what are all the possible values (aka. is it only from and to or there are also breaks), etc.

If the requirements are not clear, probably the easiest way to handle it would be to define each day of the week (Monday to Sunday) as separate `text` fields in your `companies` collection and leave the user to enter the time as free text (eventually you can specify a regex pattern if you want the field values to have a similar formatting).

Depending on the style of management you are planning, you can also move them in their own collection (eg. `schedules`) and link them with the `companies` via a `relation` field (this will make them "reusable", aka. several companies to have the exact same schedule).

**Comment by willtejeda** [comment]
another option might be to (if it's just for their schedules online ) have a few fields which could be mapped directly to individual inputs so you can style them however you'd like


weekday, start_hour, start_minute, end_hour, end_minute, (maybe an AM/PM  if you're doing 12-hour time) 

then you could also set limits to the time files so they match your format


It would probably better / cleaner to just have [weekday, start_time, end_time]  but I figure this way ^^ makes it easy to implement without too much string formatting or time conversions later.
